api/v1alpha1/common.go

```diff
@@ -0,0 +1,17 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+var (
+	// DisplayNameAnnotationKey is the key of the annotation used to set the display name of the resource
+	DisplayNameAnnotationKey = GroupVersion.Group + "/displayname"
+)
+
+// ComponetVersions Indicates the fields required for a specific version of Component.
+type ComponetVersions struct {
+	Version    string      `json:"version"`
```

Comment:
why two versions here ?
@Abirdcfly  pls add some basic filed comment

Reply:
version for helm chart version, AppVersion for app in helm chart.
for example:  the latest ingress-nginx helm chart has version 4.6.1 but appVersion 1.7.1
```
annotations:
  artifacthub.io/changes: |
    - "[helm] Support custom port configuration for internal service (#9846)"
    - "Adding resource type to default HPA configuration to resolve issues with Terraform helm chart usage (#9803)"
    - "Update Ingress-Nginx version controller-v1.7.1"
  artifacthub.io/prerelease: "false"
apiVersion: v2
appVersion: 1.7.1
description: Ingress controller for Kubernetes using NGINX as a reverse proxy and
  load balancer
home: https://github.com/kubernetes/ingress-nginx
icon: https://upload.wikimedia.org/wikipedia/commons/thumb/c/c5/Nginx_logo.svg/500px-Nginx_logo.svg.png
keywords:
- ingress
- nginx
kubeVersion: '>=1.20.0-0'
maintainers:
- name: rikatz
- name: strongjz
- name: tao12345666333
name: ingress-nginx
sources:
- https://github.com/kubernetes/ingress-nginx
version: 4.6.1
```
in  https://github.com/kubernetes/ingress-nginx/releases/tag/helm-chart-4.6.1